zerver/lib/outgoing_webhook.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import absolute_import
+from typing import Any, Iterable, Dict, Tuple, Callable, Text, Mapping
+
+import requests
+import json
+import sys
+import inspect
+import logging
+from six.moves import urllib
+from functools import reduce
+
+from django.utils.translation import ugettext as _
+
+from zerver.models import Realm, get_realm_by_email_domain, get_user_profile_by_id, get_client
+from zerver.lib.actions import check_send_message
+from zerver.lib.queue import queue_json_publish
+from zerver.lib.validator import check_dict, check_string
+from zerver.decorator import JsonableError
+
+def send_response_message(bot_id, message, response_message_content):
+    # type: (str, Dict[str, Any], Text) -> None
+    recipient_type_name = message['type']
+    bot_user = get_user_profile_by_id(bot_id)
+    realm = get_realm_by_email_domain(message['sender_email'])
+
+    if recipient_type_name == 'stream':
+        recipients = [message['display_recipient']]
+        check_send_message(bot_user, get_client("OutgoingWebhookResponse"), recipient_type_name, recipients,
+                           message['subject'], response_message_content, realm, forwarder_user_profile=bot_user)
+    else:
+        # Private message; only send if the bot is there in the recipients
+        recipients = [recipient['email'] for recipient in message['display_recipient']]
+        if bot_user.email in recipients:
+            check_send_message(bot_user, get_client("OutgoingWebhookResponse"), recipient_type_name, recipients,
+                               message['subject'], response_message_content, realm, forwarder_user_profile=bot_user)
+
+def succeed_with_message(event, success_message):
+    # type: (Dict[str, Any], Text) -> None
+    send_response_message(event['user_profile_id'], event['message'], success_message)
+
+def fail_with_message(event, failure_message):
```

Comment:
Yeah in this initial version they both are doing same operation but maybe in future they will have somewhat different operations, like maybe for `succeed_with message`, setting the content of response message as the response received from third party url. Similarly for `fail_with_message` conveying the reason for failure. 
But yeah even in this initial version they should be doing something different and your idea makes sense.